packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts

```diff
@@ -301,24 +301,21 @@ export const useGridFilter = (
       propModel: props.filterModel,
       propOnChange: props.onFilterModelChange,
       stateSelector: (state) => state.filter,
-      onChangeCallback: (model) => {
-        apiRef.current.publishEvent(GRID_FILTER_MODEL_CHANGE, model);
-      },
+      changeEvent: GRID_FILTER_MODEL_CHANGE,
     });
   }, [apiRef, props.filterModel, props.onFilterModelChange]);
 
   React.useEffect(() => {
-    const filterModel = props.filterModel;
-    if (filterModel && filterModel.items.length > 1) {
-      const hasItemsWithoutIds = filterModel.items.find((item) => item.id == null);
+    if (props.filterModel !== undefined && props.filterModel.items.length > 1) {
+      const hasItemsWithoutIds = props.filterModel.items.find((item) => item.id == null);
       if (hasItemsWithoutIds) {
         throw new Error(
           "The 'id' field is required on filterModel.items when you use multiple filters.",
         );
       }
     }
     const oldFilterModel = apiRef.current.state.filter;
```

Comment:
We could use `getState()`

Reply:
Agree, I'm merging to show a conflict with https://github.com/mui-org/material-ui-x/pull/2176/files#r680404558. We can fix this in a follow up